lib/studies/shrine/supplemental/balanced_layout.dart

```diff
@@ -226,53 +207,17 @@ List<List<Product>> balancedLayout({
   );
 
   // Check if this layout is cached.
-
   if (LayoutCache.of(context).containsKey(encodedParameters)) {
     return _generateLayout(
       products: products,
       layout: LayoutCache.of(context)[encodedParameters],
     );
   }
 
-  final List<Size> productSizes = [
-    for (var product in products)
-      _imageSize(
-        Image.asset(
-          product.assetName,
-          package: product.assetPackage,
-        ),
-      ),
-  ];
-
-  bool hasNullSize = false;
-  for (final productSize in productSizes) {
-    if (productSize == null) {
-      hasNullSize = true;
-      break;
-    }
-  }
-
-  if (hasNullSize) {
-    // If some image sizes are not read, return default layout.
-    // Default layout is not cached.
-
-    List<List<Product>> result =
-        List<List<Product>>.generate(columnCount, (columnIndex) => []);
-    for (var index = 0; index < products.length; ++index) {
-      result[index % columnCount].add(products[index]);
-    }
-
-    return result;
-  }
-
-  // All images have sizes. Use tailored layout.
-
   final List<double> productHeights = [
-    for (final productSize in productSizes)
-      productSize.height /
-              productSize.width *
-              (largeImageWidth + smallImageWidth) /
-              2 +
+    for (final product in products)
+      ((1 / product.assetAspectRatio) *
+              ((largeImageWidth + smallImageWidth) / 2)) +
```

Comment:
Some of these parens are not necessary:

1 / product.assetAspectRatio * productSize.width * (largeImageWidth + smallImageWidth) / 2

Reply:
@clocksmith 
I think it is `1 / product.assetAspectRatio * (largeImageWidth + smallImageWidth) / 2 + productCardAdditionalHeight`